sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionDescription.java

```diff
@@ -112,4 +112,5 @@
     String group() default "";
     String since() default "";
     String deprecated() default "";
+    String language() default "built-in";
```

Comment:
Hm, I think it has mixed meanings:

- `built-in` vs `hive
- `scala` vs `java` vs `python`

Also, is the second classification useful? I thought that it describes which languages support function APIs in `functions.scala`, `functions.py` and `functions.R`.

Reply:
@HyukjinKwon, thanks for reviewing.
 
Yes, it has mixed meanings. but the classification should be: `built-in`, `hive udf`, `spark udf`:
* `built-in` means the functions defined in `functions.scala`, `functions.py` and `functions.R`
* `hive udf` means the udf created by SQL `CREATE (TEMP)? FUNCTION xxx`
* `spark udf` means the udf created by `spark.udf.register()`, `spark.udf.registerJava()`, `spark.udf.registerPython()`. and the registered udfs can use `scala`, `java`, `python` languages.